src/corehost/cli/libhost.h

```diff
@@ -100,54 +107,66 @@ class corehost_init_t
+    std::vector<const pal::char_t*> m_fx_requested_versions_cstr;
+    std::vector<pal::string_t> m_fx_found_versions;
+    std::vector<const pal::char_t*> m_fx_found_versions_cstr;
 public:
     corehost_init_t(
         const pal::string_t& deps_file,
         const pal::string_t& additional_deps_serialized,
         const std::vector<pal::string_t>& probe_paths,
-        const pal::string_t& fx_dir,
         const host_mode_t mode,
-        const runtime_config_t& runtime_config)
-        : m_fx_dir(fx_dir)
-        , m_fx_name(runtime_config.get_fx_name())
-        , m_deps_file(deps_file)
+        const std::vector<fx_definition_t*>& fx_definitions)
+        : m_deps_file(deps_file)
         , m_additional_deps_serialized(additional_deps_serialized)
-        , m_portable(runtime_config.get_portable())
+        , m_portable(fx_definitions[0]->get_runtime_config().get_portable())
         , m_probe_paths(probe_paths)
-        , m_patch_roll_forward(runtime_config.get_patch_roll_fwd())
-        , m_prerelease_roll_forward(runtime_config.get_prerelease_roll_fwd())
         , m_host_mode(mode)
         , m_host_interface()
-        , m_fx_ver(runtime_config.get_fx_version())
-        , m_tfm(runtime_config.get_tfm())
+        , m_tfm(fx_definitions[0]->get_runtime_config().get_tfm())
     {
-        runtime_config.config_kv(&m_clr_keys, &m_clr_values);
         make_cstr_arr(m_clr_keys, &m_clr_keys_cstr);
         make_cstr_arr(m_clr_values, &m_clr_values_cstr);
         make_cstr_arr(m_probe_paths, &m_probe_paths_cstr);
-    }
 
-    const pal::string_t& fx_dir() const
-    {
-        return m_fx_dir;
+        int fx_count = fx_definitions.size();
+        m_fx_names.reserve(fx_count);
+        m_fx_names_cstr.reserve(fx_count);
+        m_fx_dirs.reserve(fx_count);
+        m_fx_dirs_cstr.reserve(fx_count);
+        m_fx_requested_versions.reserve(fx_count);
+        m_fx_requested_versions_cstr.reserve(fx_count);
+        m_fx_found_versions.reserve(fx_count);
+        m_fx_found_versions_cstr.reserve(fx_count);
+
+        std::unordered_map<pal::string_t, pal::string_t> combined_properties;
+        for (int i = 0; i < fx_count; ++i)
+        {
+            fx_definitions[i]->get_runtime_config().combine_properties(combined_properties);
+
+            m_fx_names.push_back(fx_definitions[i]->get_name());
+            m_fx_names_cstr.push_back(fx_definitions[i]->get_name().c_str());
+            m_fx_dirs.push_back(fx_definitions[i]->get_dir());
+            m_fx_dirs_cstr.push_back(fx_definitions[i]->get_dir().c_str());
+            m_fx_requested_versions.push_back(fx_definitions[i]->get_requested_version());
+            m_fx_requested_versions_cstr.push_back(fx_definitions[i]->get_requested_version().c_str());
+            m_fx_found_versions.push_back(fx_definitions[i]->get_found_version());
+            m_fx_found_versions_cstr.push_back(fx_definitions[i]->get_found_version().c_str());
+        }
```

Comment:
I think it would be easier to maintain if the parallel c string arrays were made with make_cstr_arr, just as with the clr_keys/values.

Reply:
I'll take a look thanks.

Reply:
Done. Thanks